pactflow/client.ts

```diff
@@ -42,16 +54,31 @@ export class PactflowClient implements Client {
     
     // PactFlow AI client methods
 
-    async generate(body: GenerationInput): Promise<GenerationResponse> {
+    /**
+     * Generate new Pact tests based on the provided input.
+     * @param toolInput The input data for the generation process.
+     * @returns The result of the generation process.
+     * @throws Error if the HTTP request fails or the operation times out.
+     */
+    async generate(toolInput: GenerationToolInput): Promise<GenerationResponse> {
       // Submit the generation request
+      if (toolInput.remoteOpenAPIDocument) {
+        const resolvedSpec = await this.resolveOpenAPISpec(toolInput.remoteOpenAPIDocument);
+        if (toolInput.body.openapi) {
+          toolInput.body.openapi.document = resolvedSpec;
+        } else {
+          throw new Error("When providing a remote OpenAPI document, the 'openapi' field must atleast have a matcher if remote openapi credentials are used.");
```

Comment:
This error message doesn't fully make sense to me. (Also, minor typo)


```suggestion
          throw new Error("When providing a remote OpenAPI document, the 'openapi' field must at least have a matcher if remote OpenAPI credentials are used.");
```

Reply:
as discussed offline i will change this with a transform/refine operation over zod

Reply:
updated.